pom.xml

```diff
@@ -10,13 +10,45 @@
     <properties>
         <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
     </properties>
-
+    <packaging>jar</packaging>
     <dependencies>
 
+        <dependency>
+            <groupId>com.sparkjava</groupId>
+            <artifactId>spark-core</artifactId>
+            <version>2.8.0</version>
+            <exclusions>
+                <exclusion>
+                    <!-- this pom will declare the version of sparkjava -->
+                    <groupId>org.slf4j</groupId>
+                    <artifactId>spark-core</artifactId>
+                </exclusion>
+            </exclusions>
+        </dependency>
+
+
+        <dependency>
+            <groupId>mysql</groupId>
+            <artifactId>mysql-connector-java</artifactId>
+            <version>8.0.18</version>
+        </dependency>
```

Comment:
Just checked and there is finally a version 8.0.20 out that supposedly deal with time conversions properly (as you might recall we had to set timezones in the URL and in code to make sure we always were doing UTC). I suspect this is less of an issue in DSM.

Reply:
Trying this out locally and it's causing some explosions with use of `beforeFirst()`.  Probably worth fixing and doing the upgrade...

```
2020-06-25 10:33:14,981 INFO  [JobScheduler_Worker-1] (org.broadinstitute.dsm.util.KitUtil.findSpecialBehaviorKits:530) - Checking pt N7TDWDSFIFV36KVADJJN for special behavior
2020-06-25 10:33:14,981 INFO  [JobScheduler_Worker-1] (org.broadinstitute.dsm.util.KitUtil.findSpecialBehaviorKits:530) - Checking pt PUI7OT7LTDRLBP18STRC for special behavior
2020-06-25 10:33:14,981 INFO  [JobScheduler_Worker-1] (org.broadinstitute.dsm.util.KitUtil.findSpecialBehaviorKits:530) - Checking pt 9C8LU8KT3OSBGTIPD0OW for special behavior
2020-06-25 10:33:14,982 INFO  [JobScheduler_Worker-1] (org.broadinstitute.dsm.util.KitUtil.findSpecialBehaviorKits:530) - Checking pt SDJZ2R73TGBHWGGNRCJ5 for special behavior
2020-06-25 10:33:15,020 INFO  [JobScheduler_Worker-1] (org.broadinstitute.ddp.db.TransactionWrapper.inTransaction:146) - retrieved connection...
2020-06-25 10:33:15,178 INFO  [JobScheduler_Worker-1] (org.broadinstitute.dsm.db.KitRequestShipping.getAllKitRequestsByRealm:390) - Collecting kit information
2020-06-25 10:33:15,217 INFO  [JobScheduler_Worker-1] (org.broadinstitute.ddp.db.TransactionWrapper.inTransaction:146) - retrieved connection...
2020-06-25 10:33:15,375 INFO  [JobScheduler_Worker-1] (org.broadinstitute.dsm.db.KitRequestShipping.getAllKitRequestsByRealm:470) - Got 0 participants kits in DSM DB for Osteo
2020-06-25 10:33:15,414 INFO  [JobScheduler_Worker-1] (org.broadinstitute.ddp.db.TransactionWrapper.inTransaction:146) - retrieved connection...
2020-06-25 10:33:15,572 ERROR [JobScheduler_Worker-1] (org.broadinstitute.dsm.jobs.DDPRequestJob.execute:58) - Some error occurred while doing pdf audit trail 
java.lang.RuntimeException: Error getting bookmark pdf_audit_kit
        at org.broadinstitute.dsm.util.DBUtil.getBookmark(DBUtil.java:33)
        at org.broadinstitute.dsm.util.PDFAudit.checkAndSavePDF(PDFAudit.java:18)
        at org.broadinstitute.dsm.jobs.DDPRequestJob.execute(DDPRequestJob.java:55)
        at org.quartz.core.JobRunShell.run(JobRunShell.java:202)
        at org.quartz.simpl.SimpleThreadPool$WorkerThread.run(SimpleThreadPool.java:573)
Caused by: java.lang.RuntimeException: Error getting bookmark pdf_audit_kit
        at org.broadinstitute.dsm.util.DBUtil.getBookmark(DBUtil.java:53)
        at org.broadinstitute.dsm.util.DBUtil.lambda$getBookmark$0(DBUtil.java:24)
        at org.broadinstitute.ddp.db.TransactionWrapper.inTransaction(TransactionWrapper.java:150)
        at org.broadinstitute.ddp.db.TransactionWrapper.inTransaction(TransactionWrapper.java:139)
        at org.broadinstitute.dsm.util.DBUtil.getBookmark(DBUtil.java:21)
        ... 4 more
Caused by: java.sql.SQLException: Operation not allowed for a result set of type ResultSet.TYPE_FORWARD_ONLY.
        at com.mysql.cj.jdbc.exceptions.SQLError.createSQLException(SQLError.java:129)
        at com.mysql.cj.jdbc.exceptions.SQLError.createSQLException(SQLError.java:97)
        at com.mysql.cj.jdbc.exceptions.SQLExceptionsMapping.translateException(SQLExceptionsMapping.java:122)
        at com.mysql.cj.jdbc.result.ResultSetImpl.last(ResultSetImpl.java:1711)
        at org.apache.commons.dbcp2.DelegatingResultSet.last(DelegatingResultSet.java:407)
        at org.apache.commons.dbcp2.DelegatingResultSet.last(DelegatingResultSet.java:407)
        at org.broadinstitute.dsm.util.DBUtil.getBookmark(DBUtil.java:43)
        ... 8 more
```